lib/cb/utils/validator.rb

```diff
@@ -31,10 +31,12 @@ def raise_response_code_errors(response)
       end
 
       def fail_with_error_details(response, error_type)
-        error = error_type.new
+        processed_response = process_response_body(response)
+        error_message = processed_response['errors'] ? processed_response['errors'][0] : nil
```

Comment:
I merged for expedience, but @Casao you are right that this should be switched. @David-Sharpe That's a great point and we likely should do that.